include/nlohmann/json.hpp

```diff
@@ -893,6 +893,12 @@ class basic_json // NOLINT(cppcoreguidelines-special-member-functions,hicpp-spec
                 JSON_ASSERT(false); // NOLINT(cert-dcl03-c,hicpp-static-assert,misc-static-assert) LCOV_EXCL_LINE
         }
         JSON_ASSERT(m_data.m_type == val.type());
+
+#if JSON_DIAGNOSTIC_POSITIONS
+        start_position = val.start_position;
+        end_position = val.end_position;
```

Comment:
@nlohmann This is a constructor, should this be in the member initialization list?  Nothing else in this constructor is in that list, but that seems to just be because they're complicated.